widgetssdk/src/main/java/com/glia/widgets/helper/Data.kt

```diff
@@ -11,6 +11,6 @@ internal sealed class Data<out T> {
 }
 
 internal fun <T : Any> Data.Companion.from(value: T?): Data<T> = when (value) {
-    null -> Data.Empty
+    null, emptyList<T>() -> Data.Empty
```

Comment:
I would prefer not to add queue-related custom logic inside the general function.
Could you please check [this](https://github.com/salemove/android-sdk-widgets/pull/1107/files#diff-866dbba53f25ac25175e7cf615a3336831fc3c62b935259f3f0c4e39830cbff4R66) line? I added this logic there.

Reply:
I was also thinking about it, however, decided that changing the from function is appropriate because:
1. At the moment function is used by `ConfigurationManager` only
2. The change itself makes sense `null, emptyList<T>() -> Data.Empty` is logical, why don't we consider an empty list as Data.Empty?

Reply:
I have several reasons against it

- The `Data` class handles possible nullability inside Rx, so `Data. Empty` means that the `Data` class/wrapper holds no value, but an empty list is a value itself and can be needed in other cases as well.
- I'd prefer to explicitly replace the empty list with null for this particular case, on the stage when we receive the queue list. That will help in the future to understand the logic easily.
- And finally, if we consider an empty list as a `Data. Empty` state, we should also consider any types of Collections, Arrays, Sets... and so on to keep this function meaningful across all these data structures.

Reply:
I removed that part from my PR since it is in this PR context.

Reply:
@DavDo Okay, I have made the changes. Please have a look now.